lib/litmus.ex

```diff
@@ -1,18 +1,54 @@
 defmodule Litmus do
   @moduledoc """
-  Documentation for Litmus.
+  Documentation for Litmus. Validate data against schema defined.
+
+  ## Examples
+
+      iex> schema = %{"id": %Litmus.Type.Any{"required": true}}
+      iex> params = %{"id": 1}
+      iex> Litmus.validate(params, schema)
+      {:ok, %{id: 1}}
+
+      iex> schema = %{"id": %Litmus.Type.Any{}}
+      iex> params = %{"password": 1}
+      iex> Litmus.validate(params, schema)
+      {:error, "password is not allowed"}
+
   """
 
+  alias Litmus.Type
+
   @doc """
-  Hello world.
+  Validate data based on a schema.
+  """
+  @spec validate(map, map) :: {:ok, map} | {:error, String.t()}
+  def validate(data, schema) do
+    case validate_allowed_params(data, schema) do
+      :ok -> validate_schema(data, schema)
+      error = {:error, _} -> error
```

Comment:
Looking at this again, I think `{:error, msg} -> {:error, msg}` is probably more readable. What do you think?

Reply:
Yeah. This way we are more clear that we are passing an atom and an error message